File: 383. Ransom Note.java
public class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] note = new int[26];
        int[] mag = new int[26];
        for (int i = 0; i < ransomNote.length(); i++) {
            note[ransomNote.charAt(i) - 'a'] += 1;
        }
        for (int j = 0; j < magazine.length(); j++) {
            mag[magazine.charAt(j) - 'a'] += 1;
        }
        for (int k = 0; k < 26; k++) {
            if (note[k] > mag[k]) {
                return false;
            }
        }
        return true;
    }
}

File: 383. Ransom Note.py
from collections import Counter
class Solution:
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        magCount = Counter(magazine)

        for letter in ransomNote:
            if letter in magCount:
                if magCount[letter] != 0:
                    magCount[letter] -= 1
                else:
                    return False
            else:
                return False
        return True
